clippy_lints/src/redundant_static_lifetime.rs

```diff
@@ -16,29 +16,32 @@ declare_clippy_lint! {
     /// ```ignore
     /// const FOO: &'static [(&'static str, &'static str, fn(&Bar) -> bool)] =
     /// &[...]
+    /// static FOO: &'static [(&'static str, &'static str, fn(&Bar) -> bool)] =
+    /// &[...]
     /// ```
     /// This code can be rewritten as
     /// ```ignore
     ///  const FOO: &[(&str, &str, fn(&Bar) -> bool)] = &[...]
+    ///  static FOO: &[(&str, &str, fn(&Bar) -> bool)] = &[...]
     /// ```
-    pub CONST_STATIC_LIFETIME,
+    pub REDUNDANT_STATIC_LIFETIME,
     style,
-    "Using explicit `'static` lifetime for constants when elision rules would allow omitting them."
+    "Using explicit `'static` lifetime for constants or statics when elision rules would allow omitting them."
 }
 
-declare_lint_pass!(StaticConst => [CONST_STATIC_LIFETIME]);
+declare_lint_pass!(RedundantStaticLifetime => [REDUNDANT_STATIC_LIFETIME]);
 
-impl StaticConst {
+impl RedundantStaticLifetime {
     // Recursively visit types
-    fn visit_type(&mut self, ty: &Ty, cx: &EarlyContext<'_>) {
+    pub fn visit_type(&mut self, ty: &Ty, cx: &EarlyContext<'_>, reason: &str) {
```

Comment:
```suggestion
    fn visit_type(&mut self, ty: &Ty, cx: &EarlyContext<'_>, reason: &str) {
```

This `pub` shouldn't be necessary.